hack/operatorhub/internal/operatorhub/operatorhub.go

```diff
@@ -0,0 +1,647 @@
+		return "", err
+	}
+	if len(images.Data) > 1 {
+		fmt.Fprintf(os.Stderr, "\nid                       creation_date                    docker_image_digest\n")
+		for _, image := range images.Data {
+			fmt.Fprintf(os.Stderr, "%s %s %s\n", image.Id, image.CreationDate, image.DockerImageDigest)
+		}
+		return "", fmt.Errorf("found %d images with tag %s in RedHat catalog while only one is expected", len(images.Data), version)
+	}
+	if len(images.Data) == 0 {
+		return "", fmt.Errorf("no image with tag %s in RedHat catalog", version)
+	}
+	imageDigest := images.Data[0].DockerImageDigest
+	if imageDigest == "" {
+		return "", fmt.Errorf("image digest for %s is empty", version)
+	}
+	return imageDigest, nil
+}
+
+// config is the configuration that matches the config.yaml
+type config struct {
+	NewVersion   string `json:"newVersion"`
+	PrevVersion  string `json:"prevVersion"`
+	StackVersion string `json:"stackVersion"`
+	CRDs         []struct {
+		Name        string `json:"name"`
+		DisplayName string `json:"displayName"`
+		Description string `json:"description"`
+	} `json:"crds"`
+	Packages []struct {
+		OutputPath          string `json:"outputPath"`
+		PackageName         string `json:"packageName"`
+		DistributionChannel string `json:"distributionChannel"`
+		OperatorRepo        string `json:"operatorRepo"`
+		UbiOnly             bool   `json:"ubiOnly"`
+		DigestPinning       bool   `json:"digestPinning"`
+	} `json:"packages"`
+}
+
+// hasDigestPinning will return true if any package
+// within the configuration has DigestPinning enabled.
+func (c *config) hasDigestPinning() bool {
+	if c == nil {
+		return false
+	}
+	for _, pkg := range c.Packages {
+		if pkg.DigestPinning {
+			return true
+		}
+	}
+	return false
+}
+
+func loadConfigFile(c GenerateConfig) (*config, error) {
+	confBytes, err := ioutil.ReadFile(c.ConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("while opening file %s: %w", c.ConfigPath, err)
+	}
+
+	var conf config
+	if err := gyaml.Unmarshal(confBytes, &conf); err != nil {
+		return nil, fmt.Errorf("while unmarshaling config from %s: %w", c.ConfigPath, err)
+	}
+
+	// allow command line flags to override config file
+	if c.NewVersion != "" {
+		conf.NewVersion = c.NewVersion
+	}
+	if c.PrevVersion != "" {
+		conf.PrevVersion = c.PrevVersion
+	}
+	if c.StackVersion != "" {
+		conf.StackVersion = c.StackVersion
+	}
+
+	return &conf, nil
+}
+
+func getInstallManifestStream(conf *config, manifestPaths []string) (io.Reader, func(), error) {
+	if len(manifestPaths) == 0 {
+		s, err := installManifestFromWeb(conf.NewVersion)
+		return s, func() {}, err
+	}
+
+	var rs []io.Reader
+	closer := func() {
+		for _, r := range rs {
+			if closer, ok := r.(io.Closer); ok {
+				closer.Close()
+			}
+		}
+	}
+	for _, p := range manifestPaths {
+		r, err := os.Open(p)
+		if err != nil {
+			return nil, closer, fmt.Errorf("while opening %s: %w", manifestPaths, err)
+		}
+		rs = append(rs, r)
```

Comment:
The fix to ensure local YAML files have a proper "end of directives" marker (#5447) is missing:
```suggestion
		rs = append(rs, r)
		// if we're using local yaml files, ensure that they have a proper
		// end of directives marker between them.
		rs = append(rs, strings.NewReader(yamlSeparator))
```